### PR TITLE
feat: bbn wallet connect

### PR DESCRIPTION
**Pull request by totraev**

### CODE REVIEW

File: src/app/state/DelegationV2State.tsx [hunk 1]
```diff
@@ -37,19 +37,19 @@ export function DelegationV2State({ children }: PropsWithChildren) {
...
+  //     if (!data?.delegations) {
+  //       return;
+  //     }
+  //     // TODO: Find the difference and update only the difference
+  //     if (!areDelegationsEqual(delegations, data.delegations)) {
+  //       setDelegations(data.delegations);
+  //     }
+  //   },
+  //   [data?.delegations, delegations, setDelegations],
+  // );
```
[the remark this answers is not in the document]
Comment (totraev) [reply src/app/state/DelegationV2State.tsx:52]: @jrwbabylonlab Yes, but atm it breaks our app

File: src/app/context/wallet/CosmosWalletProvider.tsx [hunk 2]
```diff
@@ -33,90 +35,77 @@ const CosmosWalletContext = createContext<CosmosWalletContextProps>({
...
-          errorState: ErrorState.WALLET,
-        },
-        retryAction: connectCosmos,
-      });
-      captureError(error);
-    }
-  }, [captureError, providers.cosmosProvider, showError]);
+      try {
+        const address = await provider.getAddress();
+        const client = await provider.getSigningStargateClient();
```
[the remark this answers is not in the document]
Comment (totraev) [reply src/app/context/wallet/CosmosWalletProvider.tsx:62]: I gonna delegate this question to @gbarkhatov

[the remark this answers is not in the document]
Comment (gbarkhatov) [reply src/app/context/wallet/CosmosWalletProvider.tsx:62]: @jrwbabylonlab src/app/context/wallet/WalletConnectionProvider.tsx

File: src/app/context/wallet/WalletConnectionProvider.tsx [hunk 3]
```diff
@@ -1,119 +1,51 @@
...
-import { keplrRegistry } from "@/config/wallet/babylon";
+import { Network, WalletProvider } from "@babylonlabs-io/bbn-wallet-connect";
+import { type PropsWithChildren } from "react";
 
-// We have to split the connection into two parts
-// so we can use the tomoWalletConnect and tomoModal hooks
-export const WalletConnectionProvider = ({ children }: PropsWithChildren) => {
-  const { resolvedTheme } = useTheme();
+import { ErrorState } from "@/app/types/errors";
+import { keplrRegistry } from "@/config/wallet/babylon";
```
Comment (gbarkhatov) [inline src/app/context/wallet/WalletConnectionProvider.tsx:7]: I suggest we stop using `keplrRegistry` which was a part of Tomo wallet connect and use `bbnDevnet` instead. We will also have `bbnTestnet` later on or so, so these files describe chains, `ChainInfo` in Keplr

File: package.json [hunk 4]
```diff
@@ -27,6 +27,7 @@
   "dependencies": {
     "@babylonlabs-io/babylon-proto-ts": "0.0.3-canary.3",
     "@babylonlabs-io/bbn-core-ui": "^0.2.0",
+    "@babylonlabs-io/bbn-wallet-connect": "^0.0.21",
```
Comment (gbarkhatov) [inline package.json:30]: @totraev we need to update this to 0.0.23

File: src/app/context/wallet/WalletConnectionProvider.tsx [hunk 5]
```diff
@@ -1,119 +1,51 @@
...
-  }, [tomoModal]);
-
-  const disconnect = useCallback(async () => {
-    await tomoWalletConnect.disconnect();
-  }, [tomoWalletConnect]);
+  },
+  {
+    chain: "BBN",
+    config: {
+      chainId: keplrRegistry.chainId,
```
Comment (gbarkhatov) [inline src/app/context/wallet/WalletConnectionProvider.tsx:26]: @jrwbabylonlab this is where we describe the stuff related to the chain
@totraev let's rename `keplrRegistry` to `bbnDevnet`

**Review by gbarkhatov — APPROVED** [review]

**Comment by gbarkhatov** [comment]
@totraev is it WIP change or this PR misses something from https://github.com/babylonlabs-io/simple-staking/pull/409
https://i.imgur.com/elrGdw7.png

**Comment by totraev** [comment]
@gbarkhatov it's completed on my end

**Comment by gbarkhatov** [comment]
@totraev this PR should also change the inscriptions/ordinals behavior of `simple-staking` OR have a proper return from `bbn-wallet-connect` 
Since just merging it as it is now won't enable the Staking form, so wallets are not going to be usable by anyone working on `simple-staking`

**Comment by totraev** [comment]
> @totraev this PR should also change the inscriptions/ordinals behavior of `simple-staking` OR have a proper return from `bbn-wallet-connect` Since just merging it as it is now won't enable the Staking form, so wallets are not going to be usable by anyone working on `simple-staking`

I didn't get it. Inscriptions are already a part of this PR. If you found a bug feel free to create an issue

**Pull request merged**